src/main/kotlin/io/thecontext/ci/output/TemplateRenderer.kt

```diff
@@ -0,0 +1,33 @@
+package io.thecontext.ci.output
+
+import com.github.mustachejava.DefaultMustacheFactory
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.io.StringWriter
+
+interface TemplateRenderer {
+
+    enum class Template(val resourcePath: String) {
+        Feed("feed/feed.rss.mustache"),
+        FeedEpisode("feed/feed-episode.md.mustache"),
+        WebsiteEpisode("website/episode.md.mustache"),
+    }
+
+    fun render(template: Template, contents: Map<String, Any>): String
+
+    class Impl : TemplateRenderer {
+
+        private val factory by lazy { DefaultMustacheFactory() }
+
+        override fun render(template: Template, contents: Map<String, Any>): String {
```

Comment:
NIT: If we talk about independent Template engines, then imho parameter `contents` should not be `Map<String, Any>` but rather of type Episode / prepared episode.

Reply:
Even better would be receiving a `data class` abstracting from our own datatypes...